cpp/src/arrow/flight/sql/odbc/flight_sql/config/configuration.cc

```diff
@@ -174,6 +174,20 @@ std::vector<std::string_view> Configuration::GetCustomKeys() const {
   return keys;
 }
 
+std::string Configuration::ToConnectString() const {
+  std::stringstream connect_string_buffer;
+
+  for (const auto& [key, value] : this->properties) {
+    if (value.empty()) continue;
+
+    if (value.find(' ') == std::string::npos)
```

Comment:
Note that whitespace is _not_ a character that requires escaping when passed to SQLDriverConnect. You're allowed text like DRIVER=My Driver;

You need to use the escape sequence if the value has characters that have significance in the connection string itself, such as semi-colons or equal signs,

Another question is why do we need to stringify properties into a connection string? If it's to implement a test button, we can have the test button call an internal API that takes in a map<string, string> instead. (and change SQLDriverConnect itself to break its input down into a map).

The reason for this is that there is no way to escape when curly braces are used in values in the connection string. This is an edge case but passwords might have curly braces.

Reply:
I will create a GitHub issue to let the community look into this. For our project we will just copy the string over

Reply:
Will also remove this function